src/redux/books/books.js

```diff
@@ -1,66 +1,61 @@
-import { v4 as uuidv4 } from 'uuid';
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
 // ACTION
 const ADD_BOOK = 'bookstore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
+const FETCH_BOOKS = 'bookstore/books/FETCH_BOOK';
+const URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/S9MVnP6pcqoqluhhvN3X/books';
 
-const books = [
-  {
-    id: uuidv4(),
-    title: 'Animal Farm',
-    author: 'George Orwell',
-    progress: 64,
-    chapter: 17,
-    genre: 'Animal Fable',
-  },
-
-  {
-    id: uuidv4(),
-    title: 'The Great Gatsby',
-    author: 'F.Scott Fitzgerald',
-    progress: 5,
-    chapter: 17,
-    genre: 'Tragedy',
-  },
-  {
-    id: uuidv4(),
-    title: 'Rich Dad Poor Dad',
-    author: 'Robert T.Kiyosaki',
-    progress: 20,
-    chapter: 17,
-    genre: 'Personal finance',
-  },
-  {
-    id: uuidv4(),
-    title: 'The Alchemist',
-    author: 'Paulo Coelho',
-    progress: 70,
-    chapter: 17,
-    genre: 'Quest, Adventure, Fantasy',
-  },
-];
+const books = [];
 
 // ACTION CREATORS
-export const addBook = (book) => ({
-  type: ADD_BOOK,
-  book,
-});
+export const addBook = createAsyncThunk(
+  ADD_BOOK,
+  async (book) => {
+    await axios.post(URL, {
+      item_id: book.id,
+      title: book.title,
+      author: book.author,
+      category: book.category,
+    });
+    return {
+      book: [
+        book.id,
+        [{
+          author: book.author,
+          title: book.title,
+          category: book.category,
+        }],
+      ],
+    };
+  },
+);
 
-export const removeBook = (id) => ({
-  type: REMOVE_BOOK,
-  data: id,
+export const removeBook = createAsyncThunk(REMOVE_BOOK, async (id) => {
+  await axios.delete(`${URL}/${id}`);
+  return { id };
 });
 
 // REDUCER
 const bookReducer = (state = books, action) => {
-  const index = state.findIndex((books) => books.id === action.data);
   switch (action.type) {
-    case ADD_BOOK:
-      return [...state, action.book];
-    case REMOVE_BOOK:
-      return [...state.slice(0, index), ...state.slice(index + 1)];
+    case `${FETCH_BOOKS}/fulfilled`:
+      return action.payload.books;
+    case `${ADD_BOOK}/fulfilled`:
+      return [...state, action.payload.book];
+    case `${REMOVE_BOOK}/fulfilled`:
+      return state.filter((book) => book[0] !== action.payload.id);
     default:
       return state;
   }
 };
 
+export const fetchBooks = createAsyncThunk(
+  FETCH_BOOKS,
+  async () => {
+    const { data } = await axios.get(URL);
+    return { books: Object.entries(data) };
+  },
+);
+
```

Comment:
Thunk action creator for fetching books is created 🔥

Reply:
@mateo951 
Thank you for your feedback